\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
\usepackage{setspace} 
\doublespacing

% Geometry
\usepackage{geometry}


% Use the PLoS provided bibtex style
\bibliographystyle{PLoS2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

\pagestyle{myheadings}
%% ** EDIT HERE **
%% Please insert a running head of 30 characters or less.  
%% Include it twice, once between each set of braces
\markboth{Nanopore automata}{Nanopore automata}


%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

\usepackage{setspace}
\doublespacing



\newcommand\titlestring{Nanopore automata}
\newcommand\authorstring{
Ian Holmes$^{1,2,\ast}$
\\
\textbf{1} Lawrence Berkeley National Laboratory, Berkeley, CA, USA
\\
\textbf{2} Department of Bioengineering, University of California, Berkeley, CA, USA
}

\usepackage{array}


% Labels & references for sections, figures and tables
% Comment out \secref and \seclabel for PLoS, which doesn't like numbered section references
\newcommand{\secref}[1]{Section~\ref{sec:#1}}
\newcommand{\seclabel}[1]{\label{sec:#1}}
\newcommand{\secname}[1]{``#1''}  % PLoS-style section names

% "Text S1", "Text S2", etc.
\newcommand{\supptext}[1]{Text S#1}

% "Dataset S1", "Dataset S2", etc.
\newcommand{\dataset}[1]{Dataset S#1}

% Appendix
\newcommand{\appref}[1]{Appendix~\ref{app:#1}}
\newcommand{\applabel}[1]{\label{app:#1}}

% Figure
\newcommand{\figref}[1]{Figure~\ref{fig:#1}}
\newcommand{\figlabel}[1]{\label{fig:#1}}

% Table
\newcommand{\tabnum}[1]{\ref{tab:#1}}
\newcommand{\tabref}[1]{Table~\tabnum{#1}}
\newcommand{\tablabel}[1]{\label{tab:#1}}

% Equation
\newcommand{\eqnref}[1]{Equation~\ref{eqn:#1}}
\newcommand{\eqnlabel}[1]{\label{eqn:#1}}


% need cite, check me, and other notes to self
\newcommand\needcite{{\bf [CITE]}}
\newcommand\checkme{{\bf [CHECK]}}

% indicator function
\newcommand\indicator[1]{\delta\left(#1\right)}

% Probability
\newcommand\prob[1]{\mbox{Pr} \left[ #1 \right]}

%% END MACROS SECTION

\begin{document}

% Title must be 150 words or less
\begin{flushleft}
  {\Large
    \textbf{\titlestring}
  }
\\
\authorstring
\end{flushleft}


% Table of contents
\tableofcontents


% Please keep the abstract between 250 and 300 words
%\newpage
\section{Abstract}
State machine algorithms for aligning Nanopore reads.

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLoS ONE authors please skip this step. 
% Author Summary not valid for PLoS ONE submissions.   
%\newpage
%\section{Author Summary}


%\newpage
%\section{Introduction}

%\newpage
\section{Specification}

Initial goal (Preliminary Results) is simple reusable code for aligning a segmented nanopore read (with segment currents summarized) to a reference sequence.

Longer-term goals (Specific Aims) include
\begin{itemize}
\item quasi-hierarchical series of models for processed$\to$raw data (raw, FAST5, FASTQ, FASTA)
\item transducer intersection-style models for read-pair alignment, suitable for long-read assemblers
\item systematic strategies for approximation/optimization algorithms, climbing the hierarchy (starting with k-mer or FM-index approaches)
\item transducer intersection models for aligning reads from different sequencing technologies, for improved assembly
\item transducer-based versions of Rahman \& Pachter's CGAL
\end{itemize}

\subsection{Parameterization algorithm}

Given the following inputs
\begin{itemize}
\item Reference genome (FASTA)
\item Segment-called reads (FAST5/HDF5)
\end{itemize}

Perform the following steps
\begin{itemize}
\item Perform Baum-Welch to fit a rich model
\end{itemize}

Rich model incorporates segment statistics.

\subsection{Reference search algorithm}

Given the following inputs
\begin{itemize}
\item Reference genome
\item Segment-called reads (FAST5/HDF5)
\item Parameterized rich model
\end{itemize}

Perform the following steps
\begin{itemize}
\item Perform Viterbi alignment
\end{itemize}

\subsection{Implementation}

Libraries etc.

HDF5...

\subsection{Evaluation}

Strategy...

Data sets...

\section{Methods}

Model \& inference algorithms.

\newcommand\paramlabel[1]{\mbox{\tiny #1}}

\subsection{Null model}

\begin{itemize}
\item Output alphabet: $\Re$ (real numbers signifying current readings)
\item $K$ current samples ({\em ticks}). Sequence is $y_1 \ldots y_K$
 \begin{itemize}
 \item partitioned into a sequence of $L$ {\em events}: $E_1 \ldots E_L$
 \end{itemize}
\item Parameters: $p^{\paramlabel{NullEvent}},p^{\paramlabel{NullTick}},\mu^{\paramlabel{Null}},\tau^{\paramlabel{Null}}$
\item Gaussian emissions: $y_n \sim \mbox{Normal}(\mu^{\paramlabel{Null}},\tau^{\paramlabel{Null}})$
\item Probability is
\begin{eqnarray*}
  \lefteqn{P(E_1 \ldots E_L, y_1 \ldots y_K) dy_1 \ldots dy_K} \\
& = & \left( 1 - p^{\paramlabel{NullEvent}} \right)
\prod_{\mbox{events:} E_l} p^{\paramlabel{NullEvent}} \left( 1 - p^{\paramlabel{NullTick}} \right)
\prod_{\mbox{ticks:} y_k \in E_l} p^{\paramlabel{NullTick}} P(y_k|\mu^{\paramlabel{Null}},\tau^{\paramlabel{Null}}) dy_k
\end{eqnarray*}
\end{itemize}


\subsection{Homology model}
\seclabel{HomologyModel}

\includegraphics[width=\textwidth]{figs/Transducer.pdf}

\begin{itemize}
\item Order-$N$ Mealy transducer.
\item Input alphabet: $\Omega = \{ A, C, G, T \}$ (nucleotides)
\item Output alphabet: real numbers partitioned into events, as with null model
\item States: Start, End, $\{$ Match${}_{x_1 \ldots x_N}$, Delete${}_{x_1 \ldots x_N}: x_1 \ldots x_N \in \Omega^N \}$
\item Parameters:
$p^{\paramlabel{StartEvent}}, p^{\paramlabel{BeginDelete}}, p^{\paramlabel{ExtendDelete}},$ \\
$\{ p^{\paramlabel{Skip}}_{x_1\ldots x_N}, p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}, p^{\paramlabel{MatchTick}}_{x_1\ldots x_N}, \mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N} : x_1 \ldots x_N \in \Omega^N \}$
\end{itemize}

Transducer can {\em skip} an individual base (no event emissions for that base),
or can {\em delete} a run of bases (no event emissions during the run).

The transition weights for this transducer are shown in \secref{TransducerTransitionTable}.




\subsection{Basecalling model}
\seclabel{BasecallingModel}

\includegraphics[width=\textwidth]{figs/HMM.pdf}

\begin{itemize}
\item Order-$N$ HMM.
\item Unobserved transition labels: $\Omega = \{ A, C, G, T \}$ (nucleotides)
\item Output alphabet: real numbers partitioned into events, as with null \& transducer models
\item States: Start, End, $\{$ Emit${}_{x_1 \ldots x_N}$
\item Parameters: same as transducer model, plus length parameter $p^{\paramlabel{Emit}}$,
and kmer probability distribution
  $q(x_1\ldots x_N)$
together with associated conditional distributions
  $q(x_{N+1}|x_2\ldots x_N)$
and
  $q(x_{N+1},x_{N+2}|x_2\ldots x_N) = q(x_{N+1}|x_2\ldots x_N) q(x_{N+2}|x_3\ldots x_{N+1})$.
Also define
\begin{eqnarray*}
p^{\paramlabel{LongDelete}}
& = & (p^{\paramlabel{Emit}})^N p^{\paramlabel{BeginDelete}} (p^{\paramlabel{ExtendDelete}})^{N-1} (1 - p^{\paramlabel{ExtendDelete}})
\\
p^{\paramlabel{ShortDelete}}_{x_1\ldots x_N}
& = & (p^{\paramlabel{Emit}})^2 (1 - p^{\paramlabel{BeginDelete}}) p^{\paramlabel{Skip}}_{x_1\ldots x_N}
\\
p^{\paramlabel{NoDelete}}_{x_1\ldots x_N}
& = & p^{\paramlabel{Emit}} (1 - p^{\paramlabel{BeginDelete}}) (1 - p^{\paramlabel{Skip}}_{x_1\ldots x_N})
\end{eqnarray*}
\end{itemize}

The transition weights for this HMM are shown in \secref{HMMTransitionTable}.


% Results and Discussion can be combined.
\newpage
\section{Results}




\section{Discussion}


% Do NOT remove this, even if you are not including acknowledgments
\newpage
\section{Acknowledgments}

%\section{References}
% The bibtex filename
\bibliography{../latex-inputs/alignment,../latex-inputs/reconstruction,../latex-inputs/duplication,../latex-inputs/genomics,../latex-inputs/ncrna,../latex-inputs/url}

\clearpage
\section{Figure Legends}

\clearpage
\section{Appendix}

\subsection{Exponential distribution}

\begin{eqnarray*}
x & \sim & \mbox{Exponential}(\kappa) \\
P(x|\kappa) & = & \kappa \exp(-\kappa x) \\
\mbox{E}[x] & = & \kappa^{-1} \\
\mbox{Var}[x] & = & \kappa^{-2}
\end{eqnarray*}

Rate parameter $\kappa$.


\subsection{Gamma distribution}

\begin{eqnarray*}
x & \sim & \mbox{Gamma}(\alpha,\beta) \\
P(x|\alpha,\beta) & = & \frac{x^{\alpha-1} \beta^\alpha \exp(-x \beta)}{\Gamma(\alpha)} \\
\mbox{E}[x] & = & \alpha/\beta \\
\mbox{Var}[x] & = & \alpha/\beta^2
\end{eqnarray*}

Shape parameter $\alpha$, rate parameter $\beta$.
$\Gamma()$ is the gamma function
\[
\Gamma(\alpha) = \int_0^{\infty} z^{\alpha-1} \exp(-z) dz
\]
Note $\Gamma(n) = (n-1)!$ for positive integer $n$.

\subsection{Normal distribution}

\begin{eqnarray*}
x \sim \mbox{Normal}(\mu,\tau)
\end{eqnarray*}


Mean $\mu$, precision $\tau$ (precision is reciprocal of variance).
\[
P(x|\mu,\tau)
 = \sqrt{\frac{\tau}{2\pi}} \exp \left( -\frac{\tau}{2}(x-\mu)^2 \right)
\]

\newgeometry{left=0cm}

\subsection{Transition table for nanopore transducer}
\seclabel{TransducerTransitionTable}

The following table gives the transition weights for the transducer introduced in \secref{HomologyModel}.

\small

\noindent
\begin{tabular}{lllll}
\hline
Source & Destination & Weight & Absorbs & Emits \\
\hline
Start & Start & $p^{\paramlabel{StartEvent}}$ & & $\{ y_s^{(k)}: 1 \leq k \leq K_s \}$, \\
& & $\times (p^{\paramlabel{NullTick}})^{K_s} (1-p^{\paramlabel{NullTick}})$ & & $K_s \sim \mbox{Geometric}(p^{\paramlabel{NullTick}})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_s} P(y^{(k)}_s|\mu^{\paramlabel{Start}},\tau^{\paramlabel{Start}}) dy^{(k)}_s$ & & $y_s^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Null}},\tau^{\paramlabel{Null}})$ \\
Start & Start & $1$ & $x \in \Omega$ & \\
Start & Match${}_{x_1 \ldots x_N}$ & $(1 - p^{\paramlabel{StartEvent}})$ & $x_1 \ldots x_N \in \Omega^N$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N})$ \\
Match${}_{x_1 \ldots x_N}$ & Match${}_{x_1 \ldots x_N}$ & ${p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}}$ & & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_1\ldots x_N} )$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N})$ \\
Match${}_{x_1 \ldots x_N}$ & Skip${}_{x_2 \ldots x_{N+1}}$ & $(1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}})$ & $x_{N+1} \in \Omega$ & \\
& & $\times (1 - p^{\paramlabel{BeginDelete}}) p^{\paramlabel{Skip}}_{x_2\ldots x_{N+1}}$ & & \\
Match${}_{x_1 \ldots x_N}$ & Match${}_{x_2 \ldots x_{N+1}}$ & $(1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}})$ & $x_{N+1} \in \Omega$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (1 - p^{\paramlabel{BeginDelete}}) (1 - p^{\paramlabel{Skip}}_{x_2\ldots x_{N+1}})$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}})$ \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}}) dy^{(k)}_m$ \\
Match${}_{x_1 \ldots x_N}$ & Delete${}_{x_2 \ldots x_{N+1}}$ & $(1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}})$ & $x_{N+1} \in \Omega$ & \\
& & $\times p^{\paramlabel{BeginDelete}}$ & & \\
Match${}_{x_1 \ldots x_N}$ & End & $1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}}$ & & \\
Skip${}_{x_1 \ldots x_N}$ & Skip${}_{x_2 \ldots x_{N+1}}$ & $p^{\paramlabel{Skip}}_{x_2\ldots x_{N+1}}$ & $x_{N+1} \in \Omega$ & \\
Skip${}_{x_1 \ldots x_N}$ & Match${}_{x_2 \ldots x_{N+1}}$ & $(1 - p^{\paramlabel{Skip}}_{x_2\ldots x_{N+1}})$ & $x_{N+1} \in \Omega$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}})$ \\
Delete${}_{x_1 \ldots x_N}$ & Delete${}_{x_2 \ldots x_{N+1}}$ & $p^{\paramlabel{ExtendDelete}}$ & $x_{N+1} \in \Omega$ & \\
Delete${}_{x_1 \ldots x_N}$ & Match${}_{x_2 \ldots x_{N+1}}$ & $(1 - p^{\paramlabel{ExtendDelete}})$ & $x_{N+1} \in \Omega$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}})$ \\
End & End & 1 & $x \in \Omega$ & \\
\hline
\end{tabular}

\normalsize
\restoregeometry



\newgeometry{left=0cm}

\subsection{Transition table for basecalling HMM}
\seclabel{HMMTransitionTable}

The following table gives the transition weights for the transducer introduced in \secref{BasecallingModel}.

\small

\noindent
\begin{tabular}{lllll}
\hline
Source & Destination & Weight & Unobserved & Observed \\
\hline
Start & Emit${}_{x_1 \ldots x_N}$ & $q(x_1 \ldots x_N)$ & $x_1 \ldots x_N \in \Omega^N$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N})$ \\
Emit${}_{x_1 \ldots x_N}$ & Emit${}_{x_1 \ldots x_N}$ & ${p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}}$ & & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_1\ldots x_N} )$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_1\ldots x_N})$, \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N}) dy^{(k)}_m$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_1\ldots x_N},\tau^{\paramlabel{Match}}_{x_1\ldots x_N})$ \\
Emit${}_{x_1 \ldots x_N}$ & Emit${}_{x_2 \ldots x_{N+1}}$ & $(1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}})$ & $x_{N+1} \in \Omega$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times q(x_{N+1}|x_2\ldots x_N) p^{\paramlabel{NoDelete}}_{x_2\ldots x_{N+1}}$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_2\ldots x_{N+1}})$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}})$ \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_2\ldots x_{N+1}},\tau^{\paramlabel{Match}}_{x_2\ldots x_{N+1}}) dy^{(k)}_m$ \\
Emit${}_{x_1 \ldots x_N}$ & Emit${}_{x_3 \ldots x_{N+2}}$ & $(1 - {p^{\paramlabel{MatchEvent}}_{x_1\ldots x_N}})$ & $x_{N+1} x_{N+2} \in \Omega^2$ & $\{ y_m^{(k)}: 1 \leq k \leq K_m \}$, \\
& & $\times q(x_{N+1},x_{N+2}|x_2\ldots x_N) p^{\paramlabel{ShortDelete}}_{x_2\ldots x_{N+1}}$ & & $K_m \sim \mbox{Geometric}(p^{\paramlabel{MatchTick}}_{x_3\ldots x_{N+2}})$, \\
& & $\times (p^{\paramlabel{MatchTick}}_{x_3\ldots x_{N+2}})^{K_m} (1-p^{\paramlabel{MatchTick}}_{x_3\ldots x_{N+2}})$ & & $y_m^{(k)} \sim \mbox{Normal}(\mu^{\paramlabel{Match}}_{x_3\ldots x_{N+2}},\tau^{\paramlabel{Match}}_{x_3\ldots x_{N+2}})$ \\
& & $\times \displaystyle \prod_{k=1}^{K_m} P(y^{(k)}_m|\mu^{\paramlabel{Match}}_{x_3\ldots x_{N+2}},\tau^{\paramlabel{Match}}_{x_3\ldots x_{N+2}}) dy^{(k)}_m$ \\
Emit${}_{x_1 \ldots x_N}$ & Start & $p^{\paramlabel{LongDelete}}$ & & \\
Emit${}_{x_1 \ldots x_N}$ & End & $1 - p^{\paramlabel{Emit}}$ & & \\
\hline
\end{tabular}

\normalsize
\restoregeometry

\end{document}

