\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
\usepackage{setspace} 
\doublespacing


% Use the PLoS provided bibtex style
\bibliographystyle{PLoS2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

\pagestyle{myheadings}
%% ** EDIT HERE **
%% Please insert a running head of 30 characters or less.  
%% Include it twice, once between each set of braces
\markboth{Nanopore automata}{Nanopore automata}


%% ** EDIT HERE **
%% PLEASE INCLUDE ALL MACROS BELOW

\usepackage{setspace}
\doublespacing



\newcommand\titlestring{Nanopore automata}
\newcommand\authorstring{
Ian Holmes$^{1,2,\ast}$
\\
\textbf{1} Lawrence Berkeley National Laboratory, Berkeley, CA, USA
\\
\textbf{2} Department of Bioengineering, University of California, Berkeley, CA, USA
}

\usepackage{array}


% Labels & references for sections, figures and tables
% Comment out \secref and \seclabel for PLoS, which doesn't like numbered section references
\newcommand{\secref}[1]{Section~\ref{sec:#1}}
\newcommand{\seclabel}[1]{\label{sec:#1}}
\newcommand{\secname}[1]{``#1''}  % PLoS-style section names

% "Text S1", "Text S2", etc.
\newcommand{\supptext}[1]{Text S#1}

% "Dataset S1", "Dataset S2", etc.
\newcommand{\dataset}[1]{Dataset S#1}

% Appendix
\newcommand{\appref}[1]{Appendix~\ref{app:#1}}
\newcommand{\applabel}[1]{\label{app:#1}}

% Figure
\newcommand{\figref}[1]{Figure~\ref{fig:#1}}
\newcommand{\figlabel}[1]{\label{fig:#1}}

% Table
\newcommand{\tabnum}[1]{\ref{tab:#1}}
\newcommand{\tabref}[1]{Table~\tabnum{#1}}
\newcommand{\tablabel}[1]{\label{tab:#1}}

% Equation
\newcommand{\eqnref}[1]{Equation~\ref{eqn:#1}}
\newcommand{\eqnlabel}[1]{\label{eqn:#1}}


% need cite, check me, and other notes to self
\newcommand\needcite{{\bf [CITE]}}
\newcommand\checkme{{\bf [CHECK]}}

% indicator function
\newcommand\indicator[1]{\delta\left(#1\right)}

% Probability
\newcommand\prob[1]{\mbox{Pr} \left[ #1 \right]}

%% END MACROS SECTION

\begin{document}

% Title must be 150 words or less
\begin{flushleft}
  {\Large
    \textbf{\titlestring}
  }
\\
\authorstring
\end{flushleft}


% Table of contents
\tableofcontents


% Please keep the abstract between 250 and 300 words
%\newpage
\section{Abstract}
State machine algorithms for aligning Nanopore reads.

% Please keep the Author Summary between 150 and 200 words
% Use first person. PLoS ONE authors please skip this step. 
% Author Summary not valid for PLoS ONE submissions.   
%\newpage
%\section{Author Summary}


%\newpage
%\section{Introduction}

%\newpage
\section{Specification}

Initial goal (Preliminary Results) is simple reusable code for aligning a segmented nanopore read (with segment currents summarized) to a reference sequence.

Longer-term goals (Specific Aims) include
\begin{itemize}
\item quasi-hierarchical series of models for processed$\to$raw data (raw, FAST5, FASTQ, FASTA)
\item transducer intersection-style models for read-pair alignment
\item systematic strategies for approximation/optimization algorithms, climbing the hierarchy (starting with k-mer or FM-index approaches)
\item transducer intersection models for aligning reads from different sequencing technologies
\end{itemize}

\subsection{Parameterization algorithm}

Given the following inputs
\begin{itemize}
\item Reference genome (FASTA)
\item Segment-called reads (FAST5/HDF5)
\end{itemize}

Perform the following steps
\begin{itemize}
\item Perform Baum-Welch to fit a rich model
\end{itemize}

Rich model incorporates segment statistics.

\subsection{Reference search algorithm}

Given the following inputs
\begin{itemize}
\item Reference genome
\item Segment-called reads (FAST5/HDF5)
\item Parameterized rich model
\end{itemize}

Perform the following steps
\begin{itemize}
\item Perform Viterbi alignment
\end{itemize}

\subsection{Implementation}

Libraries etc.

HDF5...

\subsection{Evaluation}

Strategy...

Data sets...

\section{Methods}

Model \& inference algorithms.

\subsection{Model}

\begin{itemize}
\item Order-$N$ transducer.
\item Input: nucleotide
\item Output: nucleotide, segment mean, duration
\item Emissions:
 \begin{itemize}
 \item categorical (base $k$-mer)
 \item mixture of Normal/gamma (mean/duration)
 \end{itemize}
\item Transitions:
 \begin{itemize}
 \item {\em Match}: emit single segment, absorb 1 base
 \item {\em Insert}: affine gap insertion of bases: emits segments, absorbs no bases
 \item {\em Delete}: affine gap deletion of bases: emits no segments, absorbs bases
 \item {\em Merge}: emit single segment, absorb 2 or 3 bases
 \item {\em Split}: emit single segment, absorb 0 bases
 \item {\em Skip}: emit single segment, absorb $2 \ldots K$ bases (large $K$, low extension penalty)
 \end{itemize}
\end{itemize}

This can be achieved by a Mealy transducer with $4 \times 4^N$ states.
The factor of $4^N$ accounts for the order-$N$ context.
For each such context, the four states are MAT, INS, DEL and SKP.

\newcommand\pgapopen{\lambda_{go}}
\newcommand\pgapextend{\lambda_{gx}}
\newcommand\pmergeopen{\lambda_{mo}}
\newcommand\pmergeextend{\lambda_{mx}}
\newcommand\psplit{\lambda_s}
\newcommand\pskipopen{\lambda_{ko}}
\newcommand\pskipextend{\lambda_{kx}}

Parameters:
 \begin{itemize}
 \item Gap opening \& extension probabilities $\pgapopen$, $\pgapextend$
 \item Merge probability $\pmergeopen$, probability that it's a 3-merge is $\pmergeextend$
 \item Split probability $\psplit$
 \item Skip probability $\pskipopen$, skip extension probability $\pskipextend$
 \end{itemize}

In general the emissions are of the form
\[
(y,m,d) \sim \mbox{CNE}(L)
\]
where $L$ is a ``label'' indexing the appropriate emission distribution
\begin{eqnarray*}
y & \sim & \mbox{Categorical}({\bf p}_L)
\\
m & \sim & \mbox{Normal}(\mu_L,\tau_L)
\\
d & \sim & \mbox{Exponential}(\kappa_L)
\end{eqnarray*}
where $\Omega$ is the nucleotide alphabet,
$y \in \Omega$ is the nucleotide as decoded by the basecaller,
$m \in \Re$ is the mean segment current, and
$d \in \Re^+$ is the segment duration.

The transition table is as follows:

\begin{tabular}{llllll}
Transition & From & To & Weight & Input & Output
\\
\hline
Match & MAT & MAT & $(1-\pgapopen)(1-\pmergeopen)(1-\psplit)(1-\pskipopen)$ & $x \in \Omega$ &
$(y,m,d) \sim \mbox{CNE}(\mbox{match},x,c)$
\\
Insert & MAT & INS & $\pgapopen/2$ & none & $(y,m,d) \sim \mbox{CNE}(\mbox{insert})$ \\
       & INS & INS & $\pgapextend$ & none & $(y,m,d) \sim \mbox{CNE}(\mbox{insert})$ \\
       & INS & MAT & $1 - \pgapextend$ & none & none \\
Delete & MAT & DEL & $\pgapopen/2$ & $x \in \Omega$ & none \\
       & DEL & DEL & $\pgapextend$ & $x \in \Omega$ & none \\
       & DEL & MAT & $1 - \pgapextend$ & none & none \\
Merge & MAT & MAT & $\pmergeopen(1-\pmergeextend)$ & $x \in \Omega^2$ & $(y,m,d) \sim \mbox{CNE}(\mbox{merge2},x)$ \\
      & MAT & MAT & $\pmergeopen\pmergeextend$ & $x \in \Omega^3$ & $(y,m,d) \sim \mbox{CNE}(\mbox{merge3},x)$ \\
Split & MAT & MAT & $\psplit$ & none & $(y,m,d) \sim \mbox{CNE}(\mbox{split},x)$ \\
Skip & MAT & SKP & $\pskipopen$ & $x \in \Omega$ & none \\
     & SKP & SKP & $\pskipextend$ & $x \in \Omega$ & none \\
     & SKP & MAT & $1 - \pskipextend$ & none & none
\end{tabular}

Here $c \in \Omega^N$ is the input context.



\subsection{Baum-Welch algorithm}

As usual.

Forward fill order: INS, DEL, SKP, MAT.

\subsection{Viterbi algorithm}

As usual.


% Results and Discussion can be combined.
\newpage
\section{Results}




\section{Discussion}


% Do NOT remove this, even if you are not including acknowledgments
\newpage
\section{Acknowledgments}

%\section{References}
% The bibtex filename
\bibliography{../latex-inputs/alignment,../latex-inputs/reconstruction,../latex-inputs/duplication,../latex-inputs/genomics,../latex-inputs/ncrna,../latex-inputs/url}

\clearpage
\section{Figure Legends}

\clearpage
\section{Appendix}

\subsection{Exponential distribution}

\begin{eqnarray*}
x & \sim & \mbox{Exponential}(\kappa) \\
P(x|\kappa) & = & \kappa \exp(-\kappa x) \\
\mbox{E}[x] & = & \kappa^{-1} \\
\mbox{Var}[x] & = & \kappa^{-2}
\end{eqnarray*}

Rate parameter $\kappa$.


\subsection{Gamma distribution}

\begin{eqnarray*}
x & \sim & \mbox{Gamma}(\alpha,\beta) \\
P(x|\alpha,\beta) & = & \frac{x^{\alpha-1} \beta^\alpha \exp(-x \beta)}{\Gamma(\alpha)} \\
\mbox{E}[x] & = & \alpha/\beta \\
\mbox{Var}[x] & = & \alpha/\beta^2
\end{eqnarray*}

Shape parameter $\alpha$, rate parameter $\beta$.
$\Gamma()$ is the gamma function
\[
\Gamma(\alpha) = \int_0^{\infty} z^{\alpha-1} \exp(-z) dz
\]
Note $\Gamma(n) = (n-1)!$ for positive integer $n$.

\subsection{Normal distribution}

\begin{eqnarray*}
x \sim \mbox{Normal}(\mu,\tau)
\end{eqnarray*}


Mean $\mu$, precision $\tau$ (precision is reciprocal of variance).
\[
P(x|\mu,\tau)
 = \sqrt{\frac{\tau}{2\pi}} \exp \left( -\frac{\tau}{2}(x-\mu)^2 \right)
\]


\end{document}

